pulser-simulation/pulser_simulation/simresults.py

```diff
@@ -540,9 +545,11 @@ def sample_state(
             # XOR betwen the original array and the flips
             # Gives an array of n_detects individual shots
             new_shots = shot_arr ^ flips
+
             # Count all the new_shots
-            detected_sample_dict += Counter(
+            new_shots_count = [
                 "".join(map(str, measured)) for measured in new_shots
-            )
+            ]
+            detected_sample_dict.update(new_shots_count)
```

Comment:
Indeed, I made a new optimization in the last commit, by using tuple instead of strings for sample_dict indexing. Please take a look.

(I removed the screenshot from the issue, since I noticed there was an error in my profiling script)